crates/topos-p2p/src/runtime/mod.rs

```diff
@@ -28,60 +40,99 @@ pub struct Runtime {
 
     /// Shutdown signal receiver from the client
     pub(crate) shutdown: mpsc::Receiver<oneshot::Sender<()>>,
+
+    /// Internal state machine of the p2p layer
+    pub(crate) state_machine: StateMachine,
+
+    pub(crate) health_status: HealthStatus,
 }
 
 mod handle_command;
 mod handle_event;
 
+/// Internal state machine of the p2p layer
+///
+/// This struct may change in the future to be more flexible and to handle more
+/// complex state transitions/representation.
+#[derive(Default)]
+pub(crate) struct StateMachine {
```

Comment:
This isn't really a state machine. It's more of a metadata container that holds information on the state of the network. I'd rename it to `NetworkState` or `ConnectivityProperties` or just `State`.